src/tools/forwarder/forwarder.cpp

```diff
@@ -0,0 +1,213 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    This tool creates a terminating QUIC proxy to forward all incoming traffic
+    to a specified target.
+
+--*/
+
+#include "msquichelper.h"
+#include "msquic.hpp"
+
+const char* Alpn;
+uint16_t FrontEndPort;
+const char* BackEndTarget;
+uint16_t BackEndPort;
+QUIC_CERTIFICATE_HASH Cert;
+bool BufferedMode = true;
+
+const MsQuicApi* MsQuic;
+MsQuicRegistration* Registration;
+MsQuicConfiguration* FrontEndConfiguration;
+MsQuicConfiguration* BackEndConfiguration;
+
+#define USAGE \
+    "Usage: quicforward <alpn> <local-port> <target-name/ip>:<target-port> <thumbprint> [0/1-buffered-mode]\n"
```

Comment:
Is it possible to make a totally transparent QUIC proxy? i.e. without specifying alpn and cert stuff on the proxy side?

Reply:
the ALPN must be the same on both side since you're proxying the same protocol. Certificate is also something you only specify on the front end (where you're the server). You can't "proxy" the backend's cert.